BungieFrameWork/BFW_Headers/BFW_BinaryData.h

```diff
@@ -67,29 +67,29 @@ typedef struct BDtMethods
 			1;														\
 			do {													\
 			(dst) = *((type *)(src));								\
-			((UUtUns8*)(src)) += 1;									\
+			*(UUtUns8**)&(src) += 1;									\
```

Comment:
As the commit message explains, casting creates an "rvalue" (like a temporary value; something like `src + 1` also creates an rvalue) and standard C does not allow rvalues to be assigned to.
Similarly an expression as in `char a; (long)a = 5;` obviously doesn't make sense. It would be conceivable to allow assignments in such cases if the size of the target type is smaller than or equal to the size of the source type, but the standard does not specify it as such, thus it is not allowed in conforming C.

Dereferencing produces an "lvalue" and lvalues are the things than can be assigned to.
A fun-fact: The standard explains "The name ‘‘lvalue’’ comes originally from the assignment expression `E1 = E2`, in which the left operand `E1` is required to be a (modifiable) lvalue.".

If you want more information on this I would recommend reading https://en.cppreference.com/w/c/language/value_category.

Reply:
Thanks for the explanation.
Sorry about not reading the commit message.

I suppose the target is to get rid of all `#pragma warning(disable` in `BFW.h`?

Reply:
No worries, it wasn't meant as a jab.

It could be that MSVC also has a warning for that, which may be disabled via `#pragma`. I've compiled this on Linux using gcc, which warned about this by default.
That said, I'm usually for adapting code in favor of disabling warnings. @hogsy will be the one to decide how to deal with it for this project though.